hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/StreamEventJournalP.java

```diff
@@ -118,49 +113,52 @@ private long getSequence(EventJournalInitialSubscriberState state) {
 
     @Override
     public boolean complete() {
-        long idleCount = 0;
-        while (!jobFuture.isDone()) {
-            boolean fetched = poll();
-            if (fetched) {
-                idleCount = 0;
-            } else {
-                IDLER.idle(++idleCount);
+        pollIfNeeded();
+        if (traverser == null) {
```

Comment:
There is too much nesting in this block of code, and the body of the `if` statement is very long. This should be simplified.